File: 4. LuckyNumberApp/app/src/main/java/com/example/luckynumberapp/MainActivity.java
package com.example.luckynumberapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {


    EditText editText;
    Button btn;
    TextView txt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        txt = findViewById(R.id.textView);
        editText = findViewById(R.id.edit_text);
        btn  = findViewById(R.id.btn);


        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String enteredName = editText.getText().toString();
                Intent intent = new Intent(MainActivity.this,MainActivity2.class);
                intent.putExtra("name",enteredName);
                startActivity(intent);
            }
        });

    }
}

File: __Kotlin Apps/Lucky Number App/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val edittext: EditText = findViewById(R.id.edit_text)
        val btn: Button = findViewById(R.id.btn)

        btn.setOnClickListener(View.OnClickListener {
            var editTextString = edittext.text.toString()

            Toast.makeText(this,editTextString,Toast.LENGTH_LONG).show()

            val intent : Intent = Intent(this,MainActivity2::class.java)
            intent.putExtra("name",editTextString)
            startActivity(intent)


        })

    }
}